forc-pkg/src/manifest.rs

```diff
@@ -240,6 +240,35 @@ impl PackageManifestFile {
         Ok(Self { manifest, path })
     }
 
+    /// Apply patch declarations from the workspace (which declares this package manifest as a
+    /// member) to this package manifest.
+    ///
+    /// This checks whether there is such workspace present before attempting to apply patches so
+    /// it is safe to use with standalone package manifests as well.
+    pub fn with_workspace_patches(&self) -> Self {
```

Comment:
Would it be possible to have a patch in both a package and a workspace at the top-level? If yes, wouldn't the workspace patch take precedence over the package patch?

For example, if at the top-level, we're patching with `my-branch`. But in a specific member, we declare another patch.

Reply:
Yes it is possible and it was intentionally done that way so that you can override any specific patch applied to individual packages from a single point (from the workspace that declares them as members). I had it documented but apparently deleted it while refactoring.

Reply:
I see, so the patches declared in the workspace take higher priority over whatever the patches declared are downstream?

Reply:
> I think it makes sense for the formatter to prioritize the package level one so that you can have specific formatting logic for each member. But at the same time, is it something that we should allow? While developing the mvp for the formatter we decided to be somewhat opinionated and does not allow a lot of configurability like the go formatter. 

Good question! Honestly, I myself am against too much configurability. I think having similar formatting enforced on a ecosystem-wide level seems better than enforcing formatting per-project (i like zig for this as well). I was under the assumption that we were going for a `rustfmt`-like experience (which tbh, I don't fancy) :sweat_smile: If it is the case that we want to push for a more opinionated formatter, I'd +1 for that, but that's a whole another topic.

As for patching, I do not really have a strong opinion on whether a package-level patch should take precedence over a workspace-level patch, it was just the order that felt natural to me, and was just asking questions to clarify my own understanding. Besides, I think it would probably be trivial to update this behavior further down the line if we find that users don't like this.